### PR TITLE
RGW - Zipper - Provide default get_new_req_id()

### PR DESCRIPTION
**Pull request by dang**
Now that RadorStore's get_new_req_id() just returns a random number, use
that as the default implementation for SAL.

Signed-off-by: Daniel Gryniewicz <dang@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_sal.h [hunk 1]
```diff
@@ -358,7 +358,9 @@ class Store {
     /** Enable or disable a set of bucket.  e.g. if a User is suspended */
     virtual int set_buckets_enabled(const DoutPrefixProvider* dpp, std::vector<rgw_bucket>& buckets, bool enabled) = 0;
     /** Get a new request ID */
-    virtual uint64_t get_new_req_id() = 0;
+    virtual uint64_t get_new_req_id() {
+      return ceph::util::generate_random_number<uint64_t>();
```
Comment (cbodley) [inline src/rgw/rgw_sal.h:362]: it's not clear how rgw_sal.h is getting this "include/random.h" include - can you please #include it explicitly?

**Comment by cbodley** [comment]
p.s. can we remove `RGWRados::get_new_req_id()` too?

**Comment by dang** [comment]
https://pulpito.ceph.com/dang-2022-05-18_09:24:47-rgw-wip-dang-zipper-req-id-distro-default-smithi/

**Comment by dang** [comment]
@cbodley This should be complete.

**Comment by dang** [comment]
Apparently I pushed to ci, rather than dang.  Pushed now.

**Pull request merged**